core/git-access.c

```diff
@@ -195,19 +195,43 @@ int credential_ssh_cb(git_cred **out,
 		  void *payload)
 {
 	(void) url;
-	(void) allowed_types;
 	(void) payload;
+	(void) username_from_url;
 
-	/* Bail out from libgit authentication loop when credentials are incorrect */
-	if (auth_attempt++ > 2) {
-		report_error("Authentication to cloud storage failed.");
-		return GIT_EUSER;
+	const char *username = prefs.cloud_storage_email_encoded;
+	const char *passphrase = prefs.cloud_storage_password ? prefs.cloud_storage_password : "";
+
+	// TODO: We need a way to differentiate between password and private key authentication
+	if (allowed_types & GIT_CREDTYPE_SSH_KEY) {
+		char *priv_key = format_string("%s/%s", system_default_directory(), "ssrf_remote.key");
+		if (!access(priv_key, F_OK)) {
+			/* Bail out from libgit authentication loop when credentials are incorrect */
+			if (auth_attempt++ > 2) {
```

Comment:
maybe we should use a re-usable macro or a constant for the `2` (max number of attempts) ?

Reply:
Factored this out into a function and made `max_auth_attempts` a static variable. For C++ I'm sure that this would be compiled out, so no different to a macro - for C I'm not sure, but will check.

Reply:
gcc will probably preserve the branch (function) without optimization flags.